client/src/Components/Dialogue/index.tsx

```diff
@@ -0,0 +1,229 @@
+/* eslint-disable react/prop-types */
+import React from 'react'
+import './index.css'
+import Button from '../Button'
+import bananaMan from './The-Dicer-Colour.gif'
+import transformMan from './Mighty-Milk.gif'
+import { useState } from 'react'
+import Typewriter from 'Components/Typewriter'
+export interface DialogueProps {
+  /**
+   * Header colour: enter either orange/primary
+   */
+  headerColor?: 'primary' | 'orange'
+
+  /**
+   * The character's name
+   */
+  characterName?: string
```

Comment:
These comments don't add anything and are actually an increased maintenance burden. If I change this prop I now also have to manually keep the comment in sync, and because it's free text there are no tools that help remind me.

The best thing to do here is to not have comments and have good descriptive variable names. Some examples:
- `characterName` is good
- `transform` is misleading, it could rather be `allowTransform` or - `showTransformButtons`. I first thought this might be a flag to determine if the character had transformed.
- `avatar` and `transformAvatar` could also instead be `avatarUrl` and `transformAvatarUrl`.
- `direction` could rather be `characterSide` or `avatarSide`